telluric/collections.py

```diff
@@ -308,8 +347,8 @@ def __init__(self, results):
 
         Parameters
         ----------
-        results : list
-            List of :py:class:`~telluric.features.GeoFeature` objects.
+        results : iterable
```

Comment:
Not really: `FeatureCollection` is turning the results to a list, always. The thing is that the docs were wrong, because the input didn't need to be a list, it could be anything. For lazy loading from disk we have `FileCollection`, and for streaming collections we will have to wait for 0.3.0 maybe? :)